### ISSUE TITLE
flake8-commas no longer maintained

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#2276]
We need to fork [flake8-commas](https://github.com/PyCQA/flake8-commas) because it is not maintained anymore.

**Comment by cruisen** [comment]
However, as said in https://github.com/PyCQA/flake8-commas#flake8-extension-to-enforce-better-comma-placement:

> Note: Black, the uncompromising Python code formatter, or add-trailing-comma can do all this comma insertion automatically. We recommend you use one of those tools instead.

So from my point of view, we are keeping sort of a redundancy here. And hence, what is the motivation to do so?


---
See also: https://github.com/DmytroLitvinov/awesome-flake8-extensions/issues/81#issuecomment-1016167528:

**Comment by sobolevn** [comment]
Not all people use `black` 🙂 
For example, I don't use it. But I still want my CI to catch this.

**Comment by cruisen** [comment]
Ok, point taken.

**Comment by PeterJCLaw** [comment]
FWIW: I've got a fork at https://github.com/PeterJCLaw/flake8-commas/ which I believe supports Python 3.12. It's based on @karamanolev's fork (they'd done 3.11 already).

**Comment by karamanolev** [comment]
> I've finally gotten around to publishing this as https://pypi.org/project/flake8-commas-x/. Maintenance is likely to be sporadic as this is mostly a self-interest thing for the moment. If there's interest from the community I'm happy to accept contributions from others, though I can't make any promises regarding responsiveness.

Thanks for updating and publishing it! We will switch to your flake8-commas-x as part of our 3.12 migration and we'd be happy to contribute further fixes if/when needed. Having it published on pypi is a huge plus, as long as PRs for blocking issues are merged and published reasonably quickly.

**Comment by PeterJCLaw** [comment]
Update: I've been accepted as a maintainer of `flake8-commas` and am working on reintegrating `flake8-commas-x` into mainline. I've published a preview at https://pypi.org/project/flake8-commas/4.0.0.dev0/. It would be great if some folks could give that a go and let me know if it's working for you. (`pip install -U flake8-commas --pre` should work to upgrade to that pre-release). That package should be functionally identical to `flake8-commas-x`.

**Comment by karamanolev** [comment]
@PeterJCLaw I gave 4.0.0.dev0 a try on our large-ish Python/Django project, the one I had to fork flake8-commas for initially. A basic sanity check passed:

- No errors on the code as it is currently.
- Removing a few random necessary commas triggered the appropriate errors.

LGTM from me.

**Comment by PeterJCLaw** [comment]
@karamanolev thanks for testing that. I've done some further testing myself on a few other large projects and I'm happy to announce that I've now released v4 of flake8-commas. This is the same code as flake8-commas-x was, aside from being renamed. I've also published a post-release for flake8-commas-x which directs users back to flake8-commas.

For anyone who had forked my repo (perhaps just @strayge & @youranl), flake8-commas-x is now in the `x-main` branch there, while the default `main` branch is in sync with https://github.com/PyCQA/flake8-commas.
